open-craft/opencraft
# BB-962: Make base playbook configurable

This PR adds support for changing the name of the base Open edX playbook. This is needed because edX recently changed the name of the base playbook from `edx_sandbox.yml` to `openedx_native.yml`. 
[More info](https://github.com/edx/configuration/pull/5025)

**Testing instructions:**
1. Using `master`, create a new OpenEdXInstance for `ginkgo`, `hawthorn`, `ironwood` and `master` on your Ocim devstack.
2. Pull this branch and apply migrations.
3. Check that all the instances that you created have `OpenEdXInstance.configuration_playbook_name` set to the correct playbook name:
* `ginkgo`: `playbooks/edx_sandbox.yml`
* `hawthorn`: `playbooks/edx_sandbox.yml`
* `ironwood`: `playbooks/openedx_native.yml`
* `master`: `playbooks/openedx_native.yml`
4. Create new instances for each release we support (`ginkgo`, `hawthorn`, `ironwood`, `master`) and leave the `configuration_playbook_name` empty, check if the correct playbook name was filled.
5. Click on **Launch New Appserver** and check if `configuration_playbook_name` was correctly set there too.
6. Check the instances on Ocim stage that were deployed for every type of deployment we support. The instances don't need to provision successfully, you just need to check if the correct playbook was used, and you can do that by checking for this line on the logs:
```
Running playbook "playbooks/openedx_native.yml" from "https://github.com/edx/configuration.git"
```
Deployed instances:
* [Master](https://stage.console.opencraft.com/instance/3280/) 
* [Hawthorn](https://stage.console.opencraft.com/instance/3290/) 
* [Ironwood](https://stage.console.opencraft.com/instance/3300/)
* [Beta test instance ](https://stage.console.opencraft.com/instance/3310/)
* [PR watcher sandbox ](https://stage.console.opencraft.com/instance/3330/)

**Reviewers:**
